version: '3'
services:
    hub:
        image: jupyterhub/jupyterhub:latest
        container_name: jupyterhub
        healthcheck:
            test: ["CMD", "curl", "-f", "http://localhost:8080"]
            interval: 30s
            timeout: 10s
            retries: 5
        ports:
            - 8085:8000
        build:
            context: ./hub
            dockerfile: Dockerfile
        volumes:
            - ./hub/jupyterhub_config.py:/srv/jupyterhub/jupyterhub_config.py:ro # config file
            - /var/run/docker.sock:/var/run/docker.sock:ro # docker socket
            - /etc/group:/etc/group:ro
            - /etc/passwd:/etc/passwd:ro
            - /etc/shadow:/etc/shadow:ro
        depends_on:
            db:
                condition: service_healthy
        networks:
            - jupyterhub
    db:
        container_name: jupyterhub-db
        healthcheck:
            test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
            interval: 10s
            timeout: 5s
            retries: 10
        image: mysql:8
        restart: always
        environment:
            - MYSQL_ROOT_PASSWORD=root
            - MYSQL_DATABASE=jupyterhub
        volumes:
            - jupyterhub-db:/var/lib/mysql
        networks:
            - jupyterhub
    lab:
        image: custom_image
        build:
            context: ./lab
            dockerfile: Dockerfile
        command: echo
volumes:
    jupyterhub-db:
networks:
    jupyterhub:
        name: jupyterhub
